effect/_intents.py

```diff
@@ -54,6 +52,50 @@ def parallel(effects):
     return Effect(ParallelEffects(list(effects)))
 
 
+@attributes(['exception', 'index'])
+class FirstError(Exception):
+    """
+    One of the effects in a :obj:`ParallelEffects` resulted in an error. This
+    represents the first such error that occurred.
+    """
+    def __str__(self):
+        return '(index=%s) %s: %s' % (
+            self.index, type(self.exception).__name__, self.exception)
+
+
+def perform_parallel_async(dispatcher, intent, box):
```

Comment:
This also isn't publicly exposed. I'm thinking of adding an effect.async module which just contains this.